cheatsheets/gleam-for-typescript-users.md

```diff
@@ -0,0 +1,827 @@
+---
+layout: page
+title: Gleam for TypeScript users
+subtitle: Hello TypeScripters!
+---
+
+- [Comments](#comments)
+- [Variables](#variables)
+- [Constants](#constants)
+- [Variables type annotations](#variables-type-annotations)
+- [Data types](#data-types) -->
+  - [Strings](#strings)
+  - [Numbers](#numbers)
+  - [Arrays and Lists](#arrays-and-lists)
+  - [Tuples](#tuples)
+  - [Destructuring Arrays and Tuples](#destructuring-arrays-and-tuples)
+  - [Objects, Records, Maps](#objects-records-maps)
+- [Functions](#functions)
+- [Blocks](#blocks)
+- [Flow control](#flow-control) -->
+  - [Conditionals](#conditionals)
+  - [Switch and Case](#switch-and-case)
+  - [Piping](#piping)
+  - [Try](#try)
+- [Modules](#modules)
+
+## Comments
+
+### TypeScript
+
+In TypeScript, comments are written with a `//` prefix.
+
+```ts
+// Hello, Joe!
+```
+
+Multi-line comments may be written like so:
+
+```ts
+/*
+ * Hello, Joe!
+ */
+```
+
+TypeScript also [supports JSDoc annotations](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html).
+
+```ts
+/**
+ * @type {string}
+ */
+var s;
+ 
+/** @type {Window} */
+var win;
+ 
+/** @type {PromiseLike<string>} */
+var promisedString;
+ 
+// You can specify an HTML Element with DOM properties
+/** @type {HTMLElement} */
+var myElement = document.querySelector(selector);
+element.dataset.myData = "";
+```
+
+Documentation blocks (docblocks) are extracted into generated API
+documentation.
```

Comment:
We don't need to teach TypeScript features here

Reply:
I was just translating to the equivalent from this cheatsheet: https://github.com/gleam-lang/website/blob/main/cheatsheets/gleam-for-php-users.md?plain=1#L58-L82

I can drop that section though.